airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -734,18 +755,37 @@ class TicketAudits(SourceZendeskSupportCursorPaginationStream):
 
     transformer = TypeTransformer(TransformConfig.DefaultSchemaNormalization)
 
-    # This endpoint uses a variant of cursor pagination with some differences from cursor pagination used in other endpoints.
-    def request_params(self, next_page_token: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
-        params = {"sort_by": self.cursor_field, "sort_order": "desc", "limit": self.page_size}
 
+class Triggers(SourceZendeskSupportCursorPaginationStream):
+    """Triggers stream: https://developer.zendesk.com/api-reference/ticketing/business-rules/triggers/#list-triggers"""
+
+    # sort_by not supported as request param when doing cursor pagination for this stream (returns DatabaseError)
```

Comment:
Not sure I get this comment. Do emit states for this stream that are not valid? If there isn't a guarantee that the state is right we shouldn't emit it as we lose data.

Reply:
When we start a sync for triggers with start_time as say 15th July 2023 it will fetch records as per this condition but these are not sorted by time. So if we keep updating state after every 100 records it will lead to data loss. So we are emitting state at the end to handle this. The state we are emitting at the end will be max of `updated_at` from all the records so this should be correct.
`get_updated_state` func logic
`
old_value = str((current_stream_state or {}).get(self.cursor_field, ""))
new_value = str((latest_record or {}).get(self.cursor_field, ""))
return {self.cursor_field: max(new_value, old_value)}
`